src/test/kotlin/org/rust/lang/core/resolve/RsTypeAwareResolveTest.kt

```diff
@@ -1024,4 +1024,26 @@ class RsTypeAwareResolveTest : RsResolveTestBase() {
             t.bar().foo(|e| { e.bar(); })
         }                      //^
     """)
+
+    fun `test associated type resolve for closure`() = checkByCode("""
+        trait A {
+            type Item;
+            fn filter<P>(self, predicate: P) -> Filter<Self, P> where Self: Sized, P: FnMut(&Self::Item) {
+                unimplemented!()
```

Comment:
There's no need in these `unimplemented!()` macro invocations in test code snippets. `fn filter<P>(...) -> ... {}` will effectively do the same work.